### PR TITLE
ML 540: added radio button activity item

### PR DESCRIPTION
**Pull request by DimitrySakalouski**

### CODE REVIEW

File: src/shared/ui/survey/RadioActivityItem/RadioActivityItem.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import RadioOption from './types';
+
+type RadioActivityItemProps = {
+  config: {
+    isOptionOrderRandomized: boolean;
+    options: Array<RadioOption>;
+    isOptionalText: boolean;
+  };
+
+  onResponseSet: (...args: any[]) => unknown;
```
Comment (BamMironov) [inline src/shared/ui/survey/RadioActivityItem/RadioActivityItem.tsx:19]: `onChange: (value: string) => void`

File: src/shared/ui/survey/RadioActivityItem/RadioActivityItem.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+};
+
+const RadioActivityItem: FC<RadioActivityItemProps> = ({
+  config,
+  onResponseSet,
+}) => {
+  const { options, isOptionOrderRandomized } = config;
+  const [radioGroupValue, setRadioGroupValue] = useState('');
+  const [optionalValue, setOoptionalValue] = useState('');
```
Comment (BamMironov) [inline src/shared/ui/survey/RadioActivityItem/RadioActivityItem.tsx:28]: setOoptionalValue - typo

File: src/shared/ui/survey/RadioActivityItem/RadioActivityItem.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+  const optionsList = useMemo(() => {
+    if (isOptionOrderRandomized) {
+      return shuffle(options);
+    }
+
+    return options;
+  }, [isOptionOrderRandomized, options]);
+
+  const handleRadioGroupValueChange = (value: string) => {
```
Comment (BamMironov) [inline src/shared/ui/survey/RadioActivityItem/RadioActivityItem.tsx:39]: handleRadioGroupValueChange => onValueChange

File: src/shared/ui/survey/RadioActivityItem/RadioItem.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+const RadioItem: FC<RadioLabelProps> = ({
+  option: { isVisible, name, value, image, description },
+}) => {
+  if (!isVisible) {
+    return null;
+  }
+
+  return (
+    <XStack width="100%" paddingVertical={5} alignItems="center">
```
Comment (BamMironov) [inline src/shared/ui/survey/RadioActivityItem/RadioItem.tsx:40]: Paddings should be set on RadioActivityItem level

Comment (BamMironov) [reply src/shared/ui/survey/RadioActivityItem/RadioItem.tsx:40]: no need to set width 100% here

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/shared/ui/survey/RadioActivityItem/RadioActivityItem.tsx (hunk 1 above)
Comment (DimitrySakalouski) [reply src/shared/ui/survey/RadioActivityItem/RadioActivityItem.tsx:19]: done

File: src/shared/ui/survey/RadioActivityItem/RadioActivityItem.tsx (hunk 2 above)
Comment (DimitrySakalouski) [reply src/shared/ui/survey/RadioActivityItem/RadioActivityItem.tsx:28]: done

File: src/shared/ui/survey/RadioActivityItem/RadioActivityItem.tsx (hunk 3 above)
Comment (DimitrySakalouski) [reply src/shared/ui/survey/RadioActivityItem/RadioActivityItem.tsx:39]: done

File: src/shared/ui/survey/RadioActivityItem/RadioItem.tsx (hunk 4 above)
Comment (DimitrySakalouski) [reply src/shared/ui/survey/RadioActivityItem/RadioItem.tsx:40]: done

**Review by BamMironov — APPROVED** [review]

**Pull request merged**